internal/ztest/clock.go

```diff
@@ -21,30 +21,167 @@
 package ztest
 
 import (
+	"container/heap"
+	"sync"
 	"time"
-
-	"github.com/benbjohnson/clock"
 )
 
-// MockClock provides control over the time.
-type MockClock struct{ m *clock.Mock }
+// MockClock is a fake source of time.
+// It implements standard time operations,
+// but allows the user to control the passage of time.
+//
+// Use the [Add] method to progress time.
+type MockClock struct {
+	mu  sync.RWMutex
+	now time.Time
 
-// NewMockClock builds a new mock clock that provides control of time.
+	// The MockClock works by maintaining a list of waiters.
+	// Each waiter knows the time at which it should be resolved.
+	// When the clock advances, all waiters that are in range are resolved
+	// in chronological order.
+	waiters waiters
+}
+
+// NewMockClock builds a new mock clock
+// using the current actual time as the initial time.
 func NewMockClock() *MockClock {
-	return &MockClock{clock.NewMock()}
+	return &MockClock{
+		now: time.Now(),
+	}
 }
 
 // Now reports the current time.
 func (c *MockClock) Now() time.Time {
-	return c.m.Now()
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	return c.now
 }
 
 // NewTicker returns a time.Ticker that ticks at the specified frequency.
+//
+// As with [time.NewTicker],
+// the ticker will drop ticks if the receiver is slow,
+// and the channel is never closed.
```

Comment:
is it worth mentioning that `Stop` won't work?

Reply:
Yeah, I'll mention Stop is a no-op.
Also note that it'll become a no-op in std in the future too ([ref](https://github.com/golang/go/issues/61542)).

Reply:
It's a little different -- with that change, `Stop` is not required for cleanup. However, `Stop` still still have a functional impact -- it stops sending ticks to the channel. With this mock implementation, `Stop` doesn't actually stop anything, so we're still writing tickets to the channel periodically

Reply:
Oh, fair. Yeah. Unfortunately, there's no way to intercept that Stop with the exported APIs. benbjohnson/clock gets around it with its own Ticker type, but I figured that's worth avoiding.